### DISCUSSION TITLE
Intercept all PropertyChanged without register callback

### DISCUSSION THREAD

**Discussion by workgroupengineering** [unoplatform/uno#7882] [Q&A] [answered]
From the tests I made, the Skia backend does not support Geometry transformations.

In a private PR I am trying to do the refactoring. In the Geometry class I need to intercept all the changes of the DependencyProperty to invalidate the Bounds.

How can I do?

In the derived classes of  Geometry , I can do the ovveraide of the OnPropertyChanged2 method, but in Geometry  it is not possible because OnPropertyChanged2 is generated by the DependencyObjectGenerator.

**Comment by workgroupengineering** [comment, marked as answer]
I found it by myself. I must use RegisterDisposablePropertyChangedCallback.

**Reply by jeromelaban** [reply]
You can use `OnPropertyChanged2` in Shape, the generator will avoid generating the method when already present. In general it's now best to use `OnPropertyChanged2` because `RegisterDisposablePropertyChangedCallback` requires delegates. `RegisterDisposablePropertyChangedCallback` should be used when observing changes for other types.
